presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -379,27 +382,40 @@ public CounterStat getHighMemorySplitSource()
                     partitions);
 
             ImmutableList.Builder<HivePartitionMetadata> results = ImmutableList.builder();
+            Map<String, Set<String>> partitionsNotReadable = new HashMap<>();
+            int unreadablePartitionsSkipped = 0;
+            int reasonsSkipped = 0;
             for (HivePartition hivePartition : partitionBatch) {
                 Partition partition = partitions.get(hivePartition.getPartitionId());
                 if (partition == null) {
                     throw new PrestoException(GENERIC_INTERNAL_ERROR, "Partition not loaded: " + hivePartition);
                 }
-                String partName = makePartName(table.getPartitionColumns(), partition.getValues());
+                String partitionName = makePartName(table.getPartitionColumns(), partition.getValues());
                 Optional<EncryptionInformation> encryptionInformation = encryptionInformationForPartitions.map(metadata -> metadata.get(hivePartition.getPartitionId()));
 
                 if (!isOfflineDataDebugModeEnabled(session)) {
                     // verify partition is online
-                    verifyOnline(tableName, Optional.of(partName), getProtectMode(partition), partition.getParameters());
+                    verifyOnline(tableName, Optional.of(partitionName), getProtectMode(partition), partition.getParameters());
 
                     // verify partition is not marked as non-readable
-                    String partitionNotReadable = partition.getParameters().get(OBJECT_NOT_READABLE);
-                    if (!isNullOrEmpty(partitionNotReadable)) {
+                    String reason = partition.getParameters().get(OBJECT_NOT_READABLE);
+                    if (!isNullOrEmpty(reason)) {
                         if (!shouldIgnoreUnreadablePartition(session) || !partition.isEligibleToIgnore()) {
-                            throw new HiveNotReadableException(tableName, Optional.of(partName), partitionNotReadable);
+                            throw new HiveNotReadableException(tableName, Optional.of(partitionName), reason);
+                        }
+                        if (partitionsNotReadable.size() < 3) {
```

Comment:
We should move `unreadablePartitionsSkipped++;` before this `if`

Reply:
@highker shouldn't unreadablePartitionsSkipped keep count of the number of unreadable partitions not displayed in the warning message (the ones which are not added to partitionsNotReadable)? You asked me before to add this variable, but that is what its function was.